### PR TITLE
fix(scripts): remove deadcode

### PR DESCRIPTION
**Pull request by shortcuts**
## 🧭 What and Why

🎟 JIRA Ticket: https://algolia.atlassian.net/browse/DI-1491

### Changes included:

leverage https://github.com/nadeesha/ts-prune to find deadcode and remove it, also added `lint:deadcode` for future usage.

### CODE REVIEW

File: scripts/package.json [hunk 1]
```diff
@@ -46,6 +47,7 @@
     "spinnies": "0.5.1",
     "ts-jest": "29.1.1",
     "ts-node": "10.9.1",
+    "ts-prune": "0.10.3",
```
Comment (millotp) [inline scripts/package.json:50]: `ts-prune` is still deprecated I don't thinks it's a good idea to add this as a dependency

Comment (shortcuts) [reply scripts/package.json:50]: yup I'm in the process of moving to an other tool but got other things to do in the meantime so... I'll leave this as a draft, unless you want to take it over?

Comment (shortcuts) [reply scripts/package.json]: I've pushed the progress to the branch in case you are interested

Comment (millotp) [reply scripts/package.json]: oh sorry I didn't check the date of the PR, no it's okay it's not urgent

**Review by millotp — COMMENTED** [review]
this is so satisfying GG !

**Review by millotp — APPROVED** [review]
nvm it's already in the scripts ahah

**Pull request merged**